### PR TITLE
feat: add roadhog + react example

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

example from dva https://github.com/dvajs/dva/tree/master/packages/dva-example-user-dashboard

- [x] roadhog should check extraBabelPlugins
- [x] replace mock api to egg api
- [x] redbox-react should be integrated to dva-hmr
- [ ] ~~replace transform-runtime to babel-preset-umi~~
- [x] map.json plugin

### CODE REVIEW

**Comment by popomore** [comment]
https://github.com/umijs/umi/pull/36

File: assets-with-roadhog/config/config.default.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  };
+
+  config.assets = {
+    devServer: {
+      command: 'roadhog dev',
+      port: 8000,
+      env: {
+        DISABLE_ESLINT: true,
+        SOCKET_SERVER: 'http://127.0.0.1:8000',
+        PUBLIC_PATH: 'http://127.0.0.1:8000/',
```
Comment (popomore) [inline assets-with-roadhog/config/config.default.js]: https://github.com/umijs/umi/pull/36 发版后这里可以改了

File: assets-with-roadhog/package.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    "autod-egg": "^1.0.0",
+    "babel-plugin-dva-hmr": "^0.4.0",
+    "babel-plugin-import": "^1.1.0",
+    "babel-plugin-transform-runtime": "^6.22.0",
+    "egg-bin": "^4.3.5",
+    "egg-ci": "^1.8.0",
+    "egg-mock": "^3.13.0",
+    "eslint": "^4.11.0",
+    "eslint-config-egg": "^5.1.0",
+    "redbox-react": "^1.5.0",
```
Comment (popomore) [inline assets-with-roadhog/package.json]: roadhog 这堆依赖还需要吗

**Comment by popomore** [comment]
等 https://github.com/sorrycc/roadhog/issues/630 之后再修改 publicPath

**Comment by fengmk2** [comment]
roadhog 搞定了，其他就是照葫芦画瓢了

File: assets-with-roadhog/test/app/controller/home.test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  before(() => {
+    mock.env('local');
+    app = mock.app();
+    return app.ready();
+  });
+  after(() => app.close());
+
+  afterEach(mock.restore);
+
+  it('should GET /', async function() {
```
Comment (fengmk2) [inline assets-with-roadhog/test/app/controller/home.test.js]: `async () => {}` 不知道那种写法好。。

**Comment by fengmk2** [comment]
不过只演示到开发阶段，生产打包模式估计很多人还会来咨询。

File: assets-with-roadhog/test/app/controller/home.test.js (hunk 3 above)
Comment (atian25) [reply assets-with-roadhog/test/app/controller/home.test.js]: 优先 arraw

**Comment by popomore** [comment]
有生产打包的，现在是放 app/public 下。

**Pull request merged**